webui-src/app/network/network.js

```diff
@@ -134,17 +135,36 @@ const FriendsList = () => {
       //)
     },
     view: () =>
-      m('.widget', [
-        m('h3', 'Friend nodes'),
-        m('hr'),
-        Object.keys(Data.gpgDetails).map(id => m(Friend, {id})),
+      m(".widget", [
+        m(".sort", {}, [
+          m("span", "Friends Node"),
+          m(
+            "select",
+            {
+              oninput: (e) => (online = e.target.value),
+              value: online,
+              onchange: online,
+            },
+            [m("option", "online"), m("option", "offline")]
+          ),
+        ]),
+
+        m("hr"),
+        Object.keys(Data.gpgDetails).map((id) => {
+          console.log(id);
+          if (online == "online" && Data.gpgDetails[id].isOnline)
+            return m(Friend, {id});
+          if (online == "offline" && !Data.gpgDetails[id].isOnline)
+            return m(Friend, {id});
+          
+        }),
```

Comment:
sorry @rottencandy  do you mean sorting filter.

Reply:
Sorry I meant to say, use [sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort). You can pass in a compare function that sorts by online status.

Also you can use [Object.entries](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries) here instead of Object.keys(), it will let you iterate with the full object.

Reply:
Thanks, @rottencandy, I will make a PR soon on it.

Reply:
@rottencandy, I have updated the changes. Please review it